core/indexer/processTransactions.go

```diff
@@ -259,6 +264,19 @@ func (tdp *txDatabaseProcessor) setTransactionSearchOrder(transactions map[strin
 	return transactions
 }
 
+func addToAlteredAddresses(tx *Transaction, alteredAddresses map[string]struct{}, miniBlock *block.MiniBlock, selfShardID uint32) {
+	if selfShardID == miniBlock.SenderShardID {
```

Comment:
you can refactor here to be more clear. 

isIntraShard ---> save sender and receiver address
isCrossShardDstMe ---> save receiver
else if isNotRewardMB --> save sender

Reply:
refactored after Iulian's suggestion